### PR TITLE
Add option to emit repeated fields as entities (#147)

### LINKED ISSUE #147: Fix-Path - Command or Function?
With the changes of https://github.com/metafacture/metafacture-fix/pull/131 it is not possible to emit the FIX internal path-structure.
While I still agree with all the changes it makes it more difficult to recreate the path-structure which is internal to fix.

For JSON/YAML this is no problem, since the flux-structure after the fix and the internal path-structure are the same.

For all formats that have repeated fields, that need dot-path refrences I cannot get possible index-numbers.

See: 
https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/master/data/testing/alma_structure/xml-structure.flux

Was:
https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/e271ce1f84d55a69e4aba67814ea663841f19437/data/testing/alma_structure/structure.txt

Now is:
https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/master/data/testing/alma_structure/structure.txt

This makes it difficult to grasp the correct path if there are repeated fields and where to add the index-numbers or arrays-wildcards.

`i.am.1.a.2.Path` in FIX would be `i.am.a.Path ` outside the FIX. 

I propose a FLUX-Function that just emits the FIX-Paths.
This would help to find path quickly.

### PR DESCRIPTION
**Pull request by fsteeg**
Resolves #147, see [Playground example](https://test.metafacture.org/playground/?flux=PG_DATA%0A%7Cdecode-xml%0A%7Chandle-generic-xml%0A%7Cfix%28%22nothing%28%29%22%2CrepeatedFieldsToEntities%3D%22true%22%29%0A%7Cflatten%0A%7Cencode-literals%0A%7Cprint%0A%3B&data=%3C%3Fxml+version%3D%221.0%22%3F%3E%0A%3Crecord%3E%0A++++%3Cfield%3Ea%3C/field%3E%0A++++%3Cfield%3Eb%3C/field%3E%0A%3C/record%3E).

### CODE REVIEW

File: metafix/src/test/java/org/metafacture/metafix/MetafixRecordTest.java [hunk 1]
```diff
@@ -2298,6 +2298,46 @@ public void transformRepeatedField() {
         );
     }
 
+    @Test
+    public void emitEntityForRepeatedField() {
+        MetafixTestHelpers.assertFix(streamReceiver, true, Arrays.asList(
```
Comment (blackwinter) [inline metafix/src/test/java/org/metafacture/metafix/MetafixRecordTest.java:2303]: Is this such a common case that we need a whole new helper for it?

We can simply call `i.setRepeatedFieldsToEntities(true)` in both tests and be done.

File: metafix/src/main/java/org/metafacture/metafix/Metafix.java [hunk 2]
```diff
@@ -199,7 +200,7 @@ public void endRecord() {
 
     private void emit(final String field, final Value value) {
         Value.asList(value, array -> {
-            final boolean isMulti = isArrayName(field);
+            final boolean isMulti = repeatedFieldsToEntities && array.size() > 1 || isArrayName(field);
```
[the remark this answers is not in the document]
Comment (fsteeg) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java]: I also wanted to add parentheses here, but my IDE reported a Checkstyle violation. The Gradle build does not complain however, even though they use the same config. So I added a `checkstyle-disable-line` comment in https://github.com/metafacture/metafacture-fix/pull/219/commits/637d93d2ca59287975bd8ee087d78ecf04dc0298.

File: metafix/src/test/java/org/metafacture/metafix/MetafixRecordTest.java (hunk 1 above)
Comment (fsteeg) [reply metafix/src/test/java/org/metafacture/metafix/MetafixRecordTest.java]: Right, thanks, good point, reverted the helper in https://github.com/metafacture/metafacture-fix/pull/219/commits/9e0724afc0040cda0986e591cdb5d3f3f8a388d3.

File: metafix/src/main/java/org/metafacture/metafix/Metafix.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (blackwinter) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java]: Hm, in that case I might withdraw my earlier comment. _If_ we're using `UnnecessaryParentheses` _and_ Checkstyle flags it, then we should probably follow (this doesn't justify an override, IMO).

But the question is why the Gradle build didn't detect it. Which Checkstyle version does your IDE use? Our build uses 8.30. Maybe there's a difference in configuration defaults. (I've verified that it still checks the file in question and that it's not accidentally ignored.)

[the remark this answers is not in the document]
Comment (fsteeg) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java:203]: My IDE plugin uses 8.44. I've updated the Gradle build to that version, it then reports the violation as well, and I've removed the parentheses: https://github.com/metafacture/metafacture-fix/pull/219/commits/51816a379564f735de85baf6dbfd3d52ebe7b19d.

[the remark this answers is not in the document]
Comment (blackwinter) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java:203]: Okay, thanks. Sorry for the trouble.

**Review by blackwinter — APPROVED** [review]

File: metafix/src/main/java/org/metafacture/metafix/Metafix.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (fsteeg) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java:203]: Oh, no, thanks for the review, I now have consistent Checkstyle setup in my IDE and in the build :+1:

**Pull request merged**